requester/requester.go

```diff
@@ -171,29 +166,39 @@ func (r *Requester) appendRecord(rec Record) {
 	}
 }
 
-func (r *Requester) refreshState() {
+// tickState refreshes the state every second.
+func (r *Requester) tickState() {
 	ticker := time.NewTicker(time.Second)
 	tick := ticker.C
 	for {
 		if r.done {
 			ticker.Stop()
 			break
 		}
-		r.printState()
+		r.updateState()
 		<-tick
 	}
 }
 
+// updateState calls r.OnStateUpdate with a new computed State
+func (r *Requester) updateState() {
+	r.safeOnStateUpdate()(r.State())
+}
```

Comment:
I am not sure I understand why we need this function instead of just calling directly `safeOnStateUpdate()`?